### ISSUE TITLE
Inference without paired protein measurement

### ISSUE THREAD

**Issue by fisherj2** [TencentAILabHealthcare/scTranslator#11]
Could you please clarify the correct way to predict protein from RNA counts only? When I tried using a query protein object containing only zeros, the stage3_inference_without_finetune.py script produced predictions that were entirely zero values.

**Comment by dproworld** [comment]
Seconding this. I'm getting 0 values as well. 

Is there a recommended way to inference on bulkseq / any examples? This is why *some* training code would be useful - especially for the bulk model.

**Comment by ziyi10** [comment]
Seconding this. I'm getting 0 values as well.

**Comment by fisherj2** [comment]
After some testing,  the following modification at line 80 of utils.py seems to fix the issue. 

`y_hat = torch.where(torch.isnan(y), torch.full_like(y_hat, 0), y_hat)`

change to 

`y_hat = torch.where(torch.isnan(y_hat), torch.full_like(y_hat, 0), y_hat)`

Seems like some upstream normalisation of the all-zero protein data causes NA values, which are used to modify the predictions. This change removed dependency between the predictions (y_hat) and the input protein expression.

**Comment by dproworld** [comment]
Ah, that makes sense.

I see you have a repo on testing protein imputation models - how does this one do relative to the ones you've tested?

Are you getting successful and reasonable predictions now? If you don't mind sharing the code you use to assemble your AnnData objects - could you? I'm struggling to properly save one that the model accepts.

**Comment by fisherj2** [comment]
Hopefully we can present the outcomes of this method alongside others in [our upcoming publication](https://www.biorxiv.org/content/10.1101/2024.12.11.627925v2)! I'm in the middle of testing scTranslator at the moment and can't tell yet how it measures against other strategies. I am building my query object like this, using the protein list from the repo.


`obs = pd.DataFrame(test_rna.obs.values.tolist(), index=test_rna.obs.index)`

read in reference protein list from repo
```
path = os.path.join(moduleDir + '/query_protein_ID.csv')
protein_list = pd.read_csv(path)  # Use their protein list
```

filter down to only the features I want to predict
```
rna_genes = test_protein.var.index
filtered_proteins = protein_list[protein_list['Hugo_Symbol'].isin(rna_genes)]
```

Create zeros object with ID metadata
```
X = np.zeros((test_rna.n_obs, filtered_proteins.shape[0]))
test_protein = ad.AnnData(X, obs=obs, var=filtered_proteins)
```

You want to make sure the 'rna_gene' variable contains the gene symbols for protein you'd like to predict.
